mipengine/controller/node_landscape_aggregator.py

```diff
@@ -114,6 +113,36 @@ def _get_node_socket_addr(node_info: NodeInfo):
     return f"{node_info.ip}:{node_info.port}"
 
 
+class DatasetsInfo(BaseModel):
+    """
+    A dictionary representation of a dataset's information.
+    Key values are the names of the datasets.
+    Values are the labels of the datasets.
+    """
+
+    values: Dict[str, str]
+
+
+class DataModelsInfo(BaseModel):
+    """
+    A dictionary representation of a data_model's information.
+    Key values are data models.
+    Values are tuples that contain datasets info and cdes for a specific data model.
+    """
+
+    values: Dict[str, Tuple[DatasetsInfo, CommonDataElements]]
+
+
+class FederationInfo(BaseModel):
+    """
+    A dictionary representation of all information for the federation.
```

Comment:
"a dictionary representation of all information for the federation" really?all information concerning the federation is represented by this dictionary? so all the information is the nodes' names and data models information? We call federation the whole system, naming this class FederationInfo gives no information about what it represents, one has to go check its code to figure out what that means